query-engine/connectors/sql-query-connector/src/ordering.rs

```diff
@@ -200,33 +197,78 @@ impl OrderByBuilder {
         order_by: &OrderByScalar,
         ctx: &Context<'_>,
     ) -> (Vec<AliasedJoin>, Column<'static>) {
-        let mut joins: Vec<AliasedJoin> = vec![];
+        let parent_alias = self.parent_alias.clone();
+        let joins: Vec<AliasedJoin> = self.compute_one2m_join(&order_by.path, parent_alias.as_ref(), ctx);
+
+        // This is the final column identifier to be used for the scalar field to order by.
+        // - If we order by a scalar field on the base model, we simply use the model's scalar field. eg:
+        //   `{modelTable}.{field}`
+        // - If there's a parent_alias, we use it to prefix the field, e.g. `{parent_alias}.{field}`
+        // - If we order by some relations, we use the alias used for the last join, e.g.
+        //   `{join_alias}.{field}`
+        let parent_table = joins
+            .last()
+            .map(|j| j.alias.to_owned())
+            .or_else(|| self.parent_alias.clone());
+        let order_by_column = order_by.field.as_column(ctx).opt_table(parent_table);
+
+        (joins, order_by_column)
+    }
 
+    pub(crate) fn compute_joins_relevance(
+        &mut self,
+        order_by: &OrderByRelevance,
+        ctx: &Context<'_>,
+    ) -> (Vec<AliasedJoin>, Expression<'static>) {
         let parent_alias = self.parent_alias.clone();
+        let joins: Vec<AliasedJoin> = self.compute_one2m_join(&order_by.path, parent_alias.as_ref(), ctx);
```

Comment:
Relations weren't taken into account at all. We're now gathering the `path` (which is an array of relation fields) that leads to the fields we try to order by on. We compute the needed joins, and then we order by the last field, aliased by the last join alias. This is the same process as regular relational ordering on scalar field.